src/index.ts

```diff
@@ -281,14 +284,20 @@ export function register (options: Options = {}): Register {
   if (typeCheck) {
     // Set the file contents into cache.
     const setCache = function (code: string, fileName: string) {
-      cache.contents[fileName] = code
-      cache.versions[fileName] = (cache.versions[fileName] + 1) || 1
+      if (cache.contents[fileName] !== code) {
+        cache.contents[fileName] = code
+        cache.versions[fileName] = (cache.versions[fileName] + 1) || 1
+      }
     }
 
     // Create the compiler host for type checking.
     const serviceHost = {
       getScriptFileNames: () => Object.keys(cache.versions),
-      getScriptVersion: (fileName: string) => String(cache.versions[fileName]),
+      getScriptVersion: (fileName: string): any => {
+        const version = cache.versions[fileName]
+        // tslint:disable-next-line
```

Comment:
Please don’t disable anything.

Reply:
I can't - it complains that `Expression is always true.`, and I don't really understand why it's always true - `version` definitely could be `undefined`, and in this case, it will be false.

Reply:
You'll need to fix the underlying type issue. It says it'll always be true because the type signature is `string`. Make it `string | undefined` instead, there may be other changes required.

Reply:
Yup, you're right, my bad. I had to fix the `cache.versions` type signature. Fixed!